src/main.cpp

```diff
@@ -3842,10 +3843,44 @@ void UnlinkPrunedFiles(std::set<int>& setFilesToPrune)
     }
 }
 
+/* This function is called from the RPC code for pruneblockchain */
+void PruneBlockFilesManual(int manualPruneHeight)
+{
+    // stuff the prune height into a global variable, and flush
+    nManualPruneHeight = (unsigned) manualPruneHeight;
+    fCheckForPruning = true;
+    CValidationState state;
+    FlushStateToDisk(state, FLUSH_STATE_NONE);
+}
+
+/* Calculate the block/rev files to delete based on height specified by user with RPC command pruneblockchain */
+void FindFilesToPruneManual(std::set<int>& setFilesToPrune)
+{
+    LOCK2(cs_main, cs_LastBlockFile);
+    if (chainActive.Tip() == NULL || pruneMode != PRUNE_MANUAL) {
+        return;
+    }
+
+    // last block to prune is the lesser of (user-specified height, MIN_BLOCKS_TO_KEEP from the tip)
+    unsigned int nLastBlockWeCanPrune = min(nManualPruneHeight, chainActive.Tip()->nHeight - MIN_BLOCKS_TO_KEEP);
+    int count=0;
+    for (int fileNumber = 0; fileNumber < nLastBlockFile; fileNumber++) {
+        if (vinfoBlockFile[fileNumber].nSize == 0 || vinfoBlockFile[fileNumber].nHeightLast > nLastBlockWeCanPrune)
+            continue;
+        PruneOneBlockFile(fileNumber);
+        setFilesToPrune.insert(fileNumber);
+        count++;
+    }
+    LogPrintf("Prune (Manual): prune_height=%d removed %d blk/rev pairs\n", nLastBlockWeCanPrune, count);
+}
+
 /* Calculate the block/rev files that should be deleted to remain under target*/
-void FindFilesToPrune(std::set<int>& setFilesToPrune, uint64_t nPruneAfterHeight)
+void FindFilesToPruneAuto(std::set<int>& setFilesToPrune)
 {
     LOCK2(cs_main, cs_LastBlockFile);
+    const CChainParams& chainparams = Params();
+    uint64_t nPruneAfterHeight = chainparams.PruneAfterHeight();
```

Comment:
We're trying to reduce calls to global Params() by passing things as arguments. This change has the opposite effect for no clear reason.